yt_dlp/extractor/bilibili.py

```diff
@@ -683,11 +682,10 @@ def _real_extract(self, url):
         aid = video_data.get('aid')
         old_video_id = format_field(aid, None, f'%s_part{part_id or 1}')
         cid = traverse_obj(video_data, ('pages', part_id - 1, 'cid')) if part_id else video_data.get('cid')
+        play_info = self._download_playinfo(video_id, cid, headers=headers, try_look=1)
```

Comment:
I'd recommend keep the original `play_info` traversal, and make a check here, so you don't have to make an unnecessary request.
```suggestion
        if not play_info or not self.is_logged_in:
            query = {'try_look': 1} if not self.is_logged_in else {}
            play_info = self._download_playinfo(video_id, cid, headers=headers, **query)
```

Reply:
`play_info` is not defined when `is_festival` is true.
When `is_festival` is False and `play_info` is None, the code can't be reached(an `ExtractorError` will be thrown either at line 657 or at line 659)
So there's no point judging `if not play_info` again.
I changed it to `if is_festival or not self.is_logged_in:` in 90f4203